components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/tfm_impl/attestation_core.c

```diff
@@ -869,7 +869,7 @@ attest_create_token(struct useful_buf_c *challenge,
     token_err = attest_token_start(&attest_token_ctx,
                                    option_flags,         /* option_flags */
                                    key_select,           /* key_select   */
-                                   COSE_ALGORITHM_ES256, /* alg_select   */
+                                   alg_select,           /* alg_select   */
```

Comment:
@ARMmbed/mbed-os-crypto This one also should be fixed in tfm repository?

Reply:
Yes, please https://git.trustedfirmware.org/trusted-firmware-m.git/tree/secure_fw/services/initial_attestation/attestation_core.c#n918

Reply:
The code in the trusted firmware is different than its version in Mbed OS.
The variable `alg_select` has been added to the Mbed OS version along with 
```
    /* Map the key select to an algorithm. Maybe someday we'll support something
    * other than ES256
   */
    switch (key_select) {
    default:
        alg_select = COSE_ALGORITHM_ES256;
    }
```
Therefore no fix is necessary upstream.

Is the file usually synchronized with the upstream version? If so, should the Mbed OS version of the file (well the function at least) be changed instead to mirror its upstream counterpart?

Reply:
OK, agreed that no change upstream is necessary.

Yes, we should take the upstream version when we replace our attestation service with the TF-M one.

In the mean time, I'd suggest removing the switch to fix the warning, as that keeps the code better aligned with upstream TF-M.

Reply:
9f44c2c